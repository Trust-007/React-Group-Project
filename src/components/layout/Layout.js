import Navbar from './Navbar';
import classes from './Layout.module.css';

const Layout = (props) => {
  const { children } = props;
  return (
    <div className={classes.container}>
      <Navbar />
      <div className={classes.children}>{children}</div>
    </div>
  );
};

export default Layout;
