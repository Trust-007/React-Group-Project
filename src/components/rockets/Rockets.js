import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import Rocket from './Rocket';
import fetchRockets from '../../redux/rockets/rocketThunk';
import classes from './Rockets.module.css';

const Rockets = () => {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchRockets());
  }, [dispatch]);
  const rockets = useSelector((state) => state.rockets);
  console.log(rockets);
  const rocketList = rockets.map((item) => (
    <Rocket
      key={item.id}
      description={item.description}
      flickrImages={item.flickr_images}
      rocketName={item.rocket_name}
      reserved={item.reserved}
      id={item.id}
    />
  ));
  let content;
  if (rocketList.length === 0) {
    content = 'No rockets Found';
  } else {
    content = rocketList;
  }
  return <div className={classes.rockets}>{content}</div>;
};

export default Rockets;
