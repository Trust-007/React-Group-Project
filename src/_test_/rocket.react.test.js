import React from 'react';

import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import fetchRockets from '../redux/rockets/rocketThunk';

import Rockets from '../components/rockets/Rockets';
import Layout from '../components/layout/Layout';

describe('display the rockets in the page', () => {
  it('should display all the rockets', () => {
    const { container } = render(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});

describe('should show the navigation', () => {
  it('full app rendering/navigating', async () => {
    render(<Layout />, { wrapper: BrowserRouter });
    expect(screen.queryByText(/Rockets/)).toMatchSnapshot();
  });
});

describe('Rocket reducers', () => {
  jest.setTimeout(30000);
  it('should fetch rockets', async () => {
    await store.dispatch(fetchRockets());
    expect(store.getState().rockets.length).toEqual(4);
  });
});
