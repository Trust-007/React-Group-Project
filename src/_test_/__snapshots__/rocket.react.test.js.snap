// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`display the rockets in the page should display all the rockets 1`] = `
<div>
  <div
    class="rockets"
  >
    No rockets Found
  </div>
</div>
`;

exports[`should show the navigation full app rendering/navigating 1`] = `
<a
  class=""
  href="/rockets"
>
  Rockets
</a>
`;
